sky/jobs/recovery_strategy.py

```diff
@@ -302,11 +302,17 @@ def _launch(self,
                 usage_lib.messages.usage.set_internal()
                 # Detach setup, so that the setup failure can be detected
                 # by the controller process (job_status -> FAILED_SETUP).
-                sky.launch(self.dag,
-                           cluster_name=self.cluster_name,
-                           detach_setup=True,
-                           detach_run=True,
-                           _is_launched_by_jobs_controller=True)
+                sky.launch(
+                    self.dag,
+                    cluster_name=self.cluster_name,
+                    # We expect to tear down the cluster as soon as the job is
+                    # finished. However, in case the controller dies, set
+                    # autodown to try and avoid a resource leak.
+                    idle_minutes_to_autostop=5,
```

Comment:
One thing we should double-check is the gap between the controller's monitoring/refreshing logic, and this idle minutes should be significantly longer than the refreshing gap. As a safety net, this autodown idle minutes might be good to set to longer, such as 10mins?

Reply:
Controller checks every 20s. Added a comment about this to the PR.